pkg/secrets/pushtofile/retrieve_secrets.go

```diff
@@ -40,7 +41,13 @@ func FetchSecretsForGroups(
 				)
 				return nil, err
 			}
-
+			if spec.ContentType == "base64" {
+				sValue, err = base64.StdEncoding.DecodeString(string(sValue))
+				if err != nil {
+					err = fmt.Errorf("failed to decode secret. Error: %v", err)
+					return nil, err
```

Comment:
In the K8s secrets implementation we are logging the decoding error as a warning here, but then still providing the original secret value. I think this makes it a bit more resilient if there is a change in the secret value to plaintext where it won't immediately break here.